From db55f5dc3f8107e677ba0e76f0c9f6e16f4af5c5 Mon Sep 17 00:00:00 2001
From: Clement Calmels <clement.calmels@intel.com>
Date: Wed, 20 Mar 2013 17:55:09 +0100
Subject: [PORT FROM MAIN] Trace when trying to set a readonly property

BZ: 94604

This patch adds a trace when trying to set a readonly value.

Change-Id: Iec6bac128cd79ef728592e1299016a552342ee2a
Orig-Change-Id: Iefa2b26e3d0f92a84ad23306f6a5430473b36608
Signed-off-by: Clement Calmels <clement.calmels@intel.com>
Reviewed-on: http://android.intel.com:8080/97748
Reviewed-by: cactus <cactus@intel.com>
Reviewed-by: Du, Chenyang <chenyang.du@intel.com>
Reviewed-by: Chen, HuaqiangX <huaqiangx.chen@intel.com>
Reviewed-by: Yao, Yong <yong.yao@intel.com>
Tested-by: Chen, HuaqiangX <huaqiangx.chen@intel.com>
Reviewed-by: buildbot <buildbot@intel.com>
Tested-by: buildbot <buildbot@intel.com>
---
 init/property_service.c |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletions(-)

diff --git a/init/property_service.c b/init/property_service.c
index 9315d22..e3c9733 100755
--- a/init/property_service.c
+++ b/init/property_service.c
@@ -353,7 +353,10 @@ int property_set(const char *name, const char *value)
 
     if(pi != 0) {
         /* ro.* properties may NEVER be modified once set */
-        if(!strncmp(name, "ro.", 3)) return -1;
+        if(!strncmp(name, "ro.", 3)) {
+            ERROR("sys_prop: unable to set read only property %s\n", name);
+            return -1;
+        }
 
         pa = __system_property_area__;
         update_prop_info(pi, value, valuelen);
-- 
1.7.4.1

